### PR TITLE
feat: make text editor selections setter async in preparation for neovim

### PR DESCRIPTION
**Pull request by saidelike**
## Checklist

- [ ] I have added [tests](https://www.cursorless.org/docs/contributing/test-case-recorder/)
- [ ] I have updated the [docs](https://github.com/cursorless-dev/cursorless/tree/main/docs) and [cheatsheet](https://github.com/cursorless-dev/cursorless/tree/main/cursorless-talon/src/cheatsheet)
- [ ] I have not broken the cheatsheet

I am making this small pr to be reviewed now. This is to make it easier to merge the neovim branch later.

### CODE REVIEW

**Comment by saidelike** [comment]
I have reverted the `selections` property as requested. 

All tests are passing now (`7082 passing (5m)`). (I think my previous failure was just due to a local change I hadn't committed.)

This is ready for review. @pokey

File: packages/cursorless-engine/src/util/setSelectionsAndFocusEditor.ts [hunk 1]
```diff
@@ -20,13 +20,15 @@ export async function setSelectionsAndFocusEditor(
   await editor.focus();
 }
 
-export function setSelectionsWithoutFocusingEditor(
+export async function setSelectionsWithoutFocusingEditor(
```
Comment (pokey) [inline packages/cursorless-engine/src/util/setSelectionsAndFocusEditor.ts:23]: worth checking to make sure this function is awaited everywhere it is called; I noticed you missed a couple `await`s on `setSelections`

**Review by pokey — COMMENTED** [review]
looks good with a couple minor comments!

File: packages/cursorless-engine/src/util/setSelectionsAndFocusEditor.ts (hunk 1 above)
Comment (saidelike) [reply packages/cursorless-engine/src/util/setSelectionsAndFocusEditor.ts:23]: Checked fine

File: packages/cursorless-vscode/src/ide/vscode/VscodeInsertSnippets.ts [hunk 2]
```diff
@@ -8,7 +8,7 @@ export async function vscodeInsertSnippet(
   ranges: Range[] | undefined,
 ): Promise<void> {
   if (ranges != null) {
-    editor.selections = ranges.map((range) => range.toSelection(false));
+    editor.setSelections(ranges.map((range) => range.toSelection(false)));
```
[the remark this answers is not in the document]
Comment (saidelike) [reply packages/cursorless-vscode/src/ide/vscode/VscodeInsertSnippets.ts]: done

File: packages/cursorless-vscode/src/ide/vscode/VscodeTextEditorImpl.ts [hunk 3]
```diff
@@ -149,7 +149,7 @@ export class VscodeTextEditorImpl implements EditableTextEditor {
 
   public async insertLineAfter(ranges?: Range[]): Promise<void> {
     if (ranges != null) {
-      this.selections = ranges.map((range) => range.toSelection(false));
+      this.setSelections(ranges.map((range) => range.toSelection(false)));
```
[the remark this answers is not in the document]
Comment (saidelike) [reply packages/cursorless-vscode/src/ide/vscode/VscodeTextEditorImpl.ts]: done

**Comment by saidelike** [comment]
After minor changes. Still ` 7082 passing (5m)` as expected. Ready for review @pokey

**Review by pokey — APPROVED** [review]

**Pull request merged**